### PR TITLE
Update to junit5 and mockito

### PR DESCRIPTION
**Pull request by chonton**
Mojo classes have many private fields changed to package-protected for unit tests to manipulate values.
junit updated from 4 to 5
jmockit migrated to mockito
multiple test frameworks removed in favor of junit5 capabilities

### CODE REVIEW

**Comment by chonton** [comment]
Rebased with buildx tests

**Comment by chonton** [comment]
OK, finally works on windows.

**Comment by rohanKanojia** [comment]
Could you please review Sonar code smells and address whichever seem appropriate to you?

**Comment by chonton** [comment]
All SQ issues removed from 'new' code.

**Comment by rohanKanojia** [comment]
Thanks, let me give it a quick review in a few days.

File: src/test/java/io/fabric8/maven/docker/service/BuildServiceTest.java [hunk 1]
```diff
@@ -264,84 +266,58 @@ public void tagImage_whenForceFalseAndNoDanglingTags_thenImageRemoved() throws D
...
     private void givenImageWithIdBeforeAndAfterTag(final String name, final String oldImageId, final String newImageId) throws DockerAccessException {
-        new Expectations() {{
-            queryService.getImageId(name);
-            result = oldImageId;
-            result = newImageId;
-            docker.getImageTags("oldimage");
-            result = Collections.emptyList();
-        }};
+        Mockito.doReturn(oldImageId, newImageId).when(queryService).getImageId(name);
+       // Mockito.doReturn(Collections.emptyList()).when(docker).getImageTags("oldimage");
```
Comment (rohanKanojia) [inline src/test/java/io/fabric8/maven/docker/service/BuildServiceTest.java:292]: why is this commented?

**Review by rohanKanojia — APPROVED** [review]

File: src/test/java/io/fabric8/maven/docker/service/BuildServiceTest.java (hunk 1 above)
Comment (chonton) [reply src/test/java/io/fabric8/maven/docker/service/BuildServiceTest.java]: excess mocking. I have removed line

File: src/test/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClientTest.java [hunk 2]
```diff
@@ -419,7 +399,11 @@ private void whenListImages() {
         }
     }
 
-    private void whenPushImage() {
+    private void whenPushImage() throws IOException {
+     /*   Mockito.verify(mockDelegate)
+            .post(Mockito.anyString(), Mockito.isNull(), Mockito.anyMap(), Mockito.any(ResponseHandler.class), Mockito.eq(200));
+
+      */
```
[the remark this answers is not in the document]
Comment (chonton) [reply src/test/java/io/fabric8/maven/docker/access/hc/DockerAccessWithHcClientTest.java:406]: already verified in method thenPushSucceeded()

File: src/test/java/io/fabric8/maven/docker/BuildMojoTest.java [hunk 3]
```diff
@@ -89,69 +87,60 @@ public void buildUsingConfiguredBuildx() throws IOException, MojoExecutionExcept
...
-        new Expectations() {{
-            serviceHub.getBuildXService();
-            result = buildXService;
-            authConfigFactory.createAuthConfig(false, false, null, null, null, null);
-            result = null;
-            dockerAccess.getNativePlatform();
-            result = "linux/amd64";
-        }};
+        Mockito.doReturn(buildXService).when(serviceHub).getBuildXService();
+        // Mockito.doReturn(null).when(authConfigFactory).createAuthConfig(false, false, null, null, null, null);
```
[the remark this answers is not in the document]
Comment (chonton) [reply src/test/java/io/fabric8/maven/docker/BuildMojoTest.java]: excess mocking

**Pull request merged**